app-frontend/src/mocks/store/mockStore.ts

```diff
@@ -0,0 +1,42 @@
+import configureStore from 'redux-mock-store';
+import { UserState } from '../../store/reducers/authorization';
+import { LinksState } from '../../store/reducers/links';
+import { SearchedLinkState } from '../../store/reducers/searchedLink';
+import { SearchedLinksState } from '../../store/reducers/searchedLinks';
+
+interface Link {
+  code?: string,
+  to?: string,
+  from?: string,
+  clicks?: number,
+  tags?: {
+    tagName: string,
+    _id: string,
+  }[],
+  description?: string,
+  _id?: string,
+  date?: Date,
+  message?: string,
+}
+
+interface LinkState {
+  data: {
+    data?: Link,
+  } | null
+  loading: boolean
+  error: string
+}
+
+export interface InitialMockState {
+  user?: UserState
+  searchedLink?: SearchedLinkState
+  searchedLinks?: SearchedLinksState
+  links?: LinksState
+  link?: LinkState
+}
+
+export function createMockStore(state: InitialMockState) { // eslint-disable-line
```

Comment:
Use your app's state type

Reply:
take a look at this construction:
https://stackoverflow.com/questions/41225891/typescript-same-derived-type-but-all-optional-keys

It'll be hard to maintain your mock store type in the way it is written now.

Reply:
should be something in the following way:

import {AppState} from '.../';

type InitialMockState = Partial<AppState>;